File: day05/py/part1.py
import re
import sys

#Read data 
text_file = open("../input.txt", "r")
lines =[v.strip("\n") for v in text_file.readlines()]

#print(lines)

doubles = re.compile(r"(.)\1") #regex for repeating characters
badstrings = re.compile('ab|cd|pq|xy') #regex for forbidden strings
nicestrings=0 #initialize counter for nice strings

for line in lines:

	#Get vowel count
	vowelcnt=0
	for letter in line:
		if letter in ['a','e','i','o','u']:
			vowelcnt+=1
	
	# Does it have doubles
	hasdoubles=re.search(doubles, line)!=None
	
	# Does it have a forbidden string
	hasbadstring=badstrings.search(line)!=None
	
	if vowelcnt>=3 and hasdoubles==False and hasbadstring==False:
		nicestrings+=1

print("Keepers:",nicestrings)

File: day05/jl/part1.jl
ret = split(read("../input.txt",String),"\n")
#ret=["ugknbfddgicrmopn","aaa","jchzalrnumimnmhp","haegwjzuvuyypxyu","dvszwmarrgswjxmb"]

doubles =r"(.)\1"
badstrings=r"ab|cd|pq|xy"

nicestrings=0
for (index, line) in enumerate(ret)
	#Get vowel count
	global vowelcnt=0
	for letter in line
		if string(letter) in ["a","e","i","o","u"]
			global vowelcnt+=1
		end
	end

	global hasdoubles = match(doubles, string(line)) != nothing
	global hasbadstring = match(badstrings, string(line)) != nothing

	println("$line $vowelcnt $hasdoubles $hasbadstring")

	if vowelcnt>=3 && hasdoubles==true && hasbadstring==false #Single & gives the wrong answer!
		global nicestrings+=1
	end
end

print("Final answer: $nicestrings")


